src/transforms/add_fields.rs

```diff
@@ -97,12 +104,27 @@ impl Transform for AddFields {
                 .into(),
                 TemplateOrValue::Value(v) => v,
             };
-            if let Some(_) = event.as_mut_log().insert(&key, value) {
-                debug!(
-                    message = "Field overwritten",
-                    field = key.as_ref(),
-                    rate_limit_secs = 30,
-                )
+            match self.r#override {
```

Comment:
minor nit: probably a good case for an if statement since you're matching on exactly two cases.

Reply:
😞